lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -453,21 +476,55 @@ static SILValue reapplyFunctionConversion(
   llvm_unreachable("Unhandled function conversion instruction");
 }
 
-/// Emits a reference to a derivative function of `original`, differentiated
-/// with respect to a superset of `desiredIndices`. Returns the `SILValue` for
-/// the derivative function and the actual indices that the derivative function
-/// is with respect to.
-///
-/// Returns `None` on failure, signifying that a diagnostic has been emitted
-/// using `invoker`.
-static std::optional<std::pair<SILValue, AutoDiffConfig>>
-emitDerivativeFunctionReference(
-    DifferentiationTransformer &transformer, SILBuilder &builder,
-    const AutoDiffConfig &desiredConfig, AutoDiffDerivativeFunctionKind kind,
-    SILValue original, DifferentiationInvoker invoker,
-    SmallVectorImpl<AllocStackInst *> &newBuffersToDealloc) {
-  ADContext &context = transformer.getContext();
+SILFunction *DifferentiationTransformer::getUnwrappedCurryThunkFunction(
+    SILFunction *originalFn) {
+  auto *abstractCE = originalFn->getDeclRef().getAbstractClosureExpr();
+  if (abstractCE == nullptr)
+    return nullptr;
+  auto *autoCE = dyn_cast<AutoClosureExpr>(abstractCE);
```

Comment:
nit: could simplify a bit with `dyn_cast_or_null`

Reply:
Applied the suggestion, thanks, see c674f80962108686e5fecd557e040f449c64dcfa